### ISSUE TITLE
[Bug] RuntimeException: java.io.EOFException caused by PhpClassReferenceProvider.getReferencesByElement(PhpClassReferenceProvider.java:72)

### ISSUE THREAD

**Issue by duymta97** [magento/magento2-phpstorm-plugin#733]
**Describe the bug** (*)

```
java.lang.RuntimeException: java.io.EOFException
	at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:132)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSConnection.handleError(PersistentFSConnection.java:291)
	at com.intellij.openapi.vfs.newvfs.persistent.FSRecords.handleError(FSRecords.java:717)
	at com.intellij.openapi.vfs.newvfs.persistent.FSRecords.readAndHandleErrors(FSRecords.java:251)
	at com.intellij.openapi.vfs.newvfs.persistent.FSRecords.readAttributeWithLock(FSRecords.java:604)
	at com.intellij.util.indexing.IndexingStamp.createOrGetTimeStamp(IndexingStamp.java:234)
	at com.intellij.util.indexing.IndexingStamp.getNontrivialFileIndexedStates(IndexingStamp.java:262)
	at com.intellij.util.indexing.FileBasedIndexImpl.doInvalidateIndicesForFile(FileBasedIndexImpl.java:1721)
	at com.intellij.util.indexing.events.ChangedFilesCollector.lambda$processFilesToUpdateInReadAction$4(ChangedFilesCollector.java:258)
	at com.intellij.util.indexing.events.ChangedFilesCollector.lambda$processFilesInReadAction$5(ChangedFilesCollector.java:283)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:705)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:647)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:223)
	at com.intellij.util.indexing.events.ChangedFilesCollector.lambda$processFilesInReadAction$6(ChangedFilesCollector.java:282)
	at com.intellij.util.ConcurrencyUtil.withLock(ConcurrencyUtil.java:234)
	at com.intellij.util.indexing.events.ChangedFilesCollector.lambda$processFilesInReadAction$7(ChangedFilesCollector.java:280)
	at com.intellij.util.indexing.events.VfsEventsMerger.processChanges(VfsEventsMerger.java:100)
	at com.intellij.util.indexing.events.ChangedFilesCollector.processFilesInReadAction(ChangedFilesCollector.java:279)
	at com.intellij.util.indexing.events.ChangedFilesCollector.processFilesToUpdateInReadAction(ChangedFilesCollector.java:254)
	at com.intellij.util.indexing.events.ChangedFilesCollector.ensureUpToDate(ChangedFilesCollector.java:215)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:755)
	at com.intellij.psi.stubs.StubIndexImpl.getContainingIds(StubIndexImpl.java:501)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:305)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:100)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:88)
	at com.jetbrains.php.PhpIndexImpl.getElements(PhpIndexImpl.java:574)
	at com.jetbrains.php.PhpIndexImpl.getByName(PhpIndexImpl.java:992)
	at com.jetbrains.php.PhpIndexImpl.getByFQN(PhpIndexImpl.java:967)
	at com.jetbrains.php.PhpIndexImpl.getClassesByFQNInternal(PhpIndexImpl.java:1042)
	at com.jetbrains.php.PhpIndexImpl.getAnyByFQN(PhpIndexImpl.java:1079)
	at com.magento.idea.magento2plugin.reference.provider.PhpClassReferenceProvider.getReferencesByElement(PhpClassReferenceProvider.java:72)
	at com.magento.idea.magento2plugin.reference.provider.CompositeReferenceProvider.getReferencesByElement(CompositeReferenceProvider.java:30)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.getReferences(ReferenceProvidersRegistryImpl.java:199)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.mapNotEmptyReferencesFromProviders(ReferenceProvidersRegistryImpl.java:162)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.doGetReferencesFromProviders(ReferenceProvidersRegistryImpl.java:141)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:43)
	at com.intellij.psi.impl.source.xml.XmlTokenImpl.getReferences(XmlTokenImpl.java:63)
	at com.intellij.psi.impl.SharedPsiElementImplUtil.addReferences(SharedPsiElementImplUtil.java:57)
	at com.intellij.psi.impl.SharedPsiElementImplUtil.findReferenceAt(SharedPsiElementImplUtil.java:33)
	at com.intellij.psi.MultiplePsiFilesPerDocumentFileViewProvider.findReferenceAt(MultiplePsiFilesPerDocumentFileViewProvider.java:184)
	at com.intellij.psi.impl.source.PsiFileImpl.findReferenceAt(PsiFileImpl.java:528)
	at com.intellij.codeInsight.TargetElementUtilBase.findReference(TargetElementUtilBase.java:189)
	at com.intellij.codeInsight.TargetElementUtilBase.doGetReferenceOrReferencedElement(TargetElementUtilBase.java:160)
	at com.intellij.codeInsight.TargetElementUtilBase.getReferenceOrReferencedElement(TargetElementUtilBase.java:204)
	at com.intellij.codeInsight.TargetElementUtilBase.getReferencedElement(TargetElementUtilBase.java:248)
	at com.intellij.codeInsight.TargetElementUtilBase.doFindTargetElement(TargetElementUtilBase.java:227)
	at com.intellij.codeInsight.TargetElementUtilBase.findTargetElement(TargetElementUtilBase.java:292)
	at com.intellij.codeInsight.TargetElementUtil.findTargetElement(TargetElementUtil.java:147)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorPsiDataProvider.getPsiElementIn(TextEditorPsiDataProvider.java:226)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorPsiDataProvider.getSlowData(TextEditorPsiDataProvider.java:139)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorPsiDataProvider.lambda$getData$0(TextEditorPsiDataProvider.java:93)
	at com.intellij.ide.impl.DataManagerImpl.getSlowData(DataManagerImpl.java:138)
	at com.intellij.ide.impl.DataManagerImpl.lambda$getDataRule$2(DataManagerImpl.java:106)
	at com.intellij.ide.impl.DataManagerImpl.lambda$getDataRule$4(DataManagerImpl.java:111)
	at com.intellij.ide.impl.DataManagerImpl.getDataFromProvider(DataManagerImpl.java:72)
	at com.intellij.ide.impl.DataManagerImpl.getDataFromProvider(DataManagerImpl.java:51)
	at com.intellij.ide.impl.DataManagerImpl.lambda$getDataFromProvider$1(DataManagerImpl.java:72)
	at com.intellij.openapi.actionSystem.DataKey.getData(DataKey.java:68)
	at com.intellij.ide.impl.dataRules.NavigatableRule.getData(NavigatableRule.java:39)
	at com.intellij.ide.impl.DataManagerImpl.getRulesData(DataManagerImpl.java:124)
	at com.intellij.ide.impl.DataManagerImpl.lambda$getDataRule$4(DataManagerImpl.java:112)
	at com.intellij.ide.impl.DataManagerImpl.getDataFromProvider(DataManagerImpl.java
```

**To Reproduce** (*)

**Expected behavior** (*)

**Screenshots**

**Please complete the following information:** (*)

 - OS: Linux 5.4.0-77-generic
 - PhpStorm/Intellij version: 2021.2.3
 - Plugin Version: 4.1.0

**Additional context**

**Comment by bohdan-harniuk** [comment]
Hello, @duymta97 !
Thank you for your report!

We'll try to fix this issue in the next bug-fix release.
Could you please assist me in the bug reproducing?

Thank you in advance!

**Comment by bohdan-harniuk** [comment]
Hello, @duymta97!

I am waiting for your reply. I cannot proceed with the fixing without your assisting.

Thank you in advance!

**Comment by bohdan-harniuk** [comment]
Hello, @duymta97!

It seems that it was already fixed in the #666.
Please, update your plugin to the latest version.

Thanks, 
Bohdan
